### ISSUE TITLE
bug: Ionic React "hardwarebackbutton" not allowing me to prevent default action.

### ISSUE THREAD

**Issue by niallduffy12** [ionic-team/ionic-framework#24243]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [X] v6.x

### Current Behavior

I have created a function to close a modal open on a selected page. The function works and closes the modal but it then navigates back the "/". It says in the documentation to use 

`event.preventDeault()`

but this doesn't seem to work at all and the backbutton still goes back to the root.

### Expected Behavior

If the modal is open stop the back button from navigating back the root of the application.

### Steps to Reproduce

document.addEventListener('backbutton', (event) => {
        if(modalIsOpen === true) {
            setIsOpen(false);
            event.preventDefault();
        }
    });

### Ionic Info

Ionic:

   Ionic CLI : 6.16.3

Utility:

   cordova-res : not installed globally
   native-run  : not installed globally

System:

   NodeJS : v10.13.0
   npm    : 6.4.1
   OS     : macOS Mojave

### Additional Information

Stack tracing also not working. previous page in the stack doesn't get triggered on back button. instead it goes to the root of the application.

**Comment by liamdebeasi** [comment]
Thanks for the issue. Ionic will automatically close modals for you before navigating back, so you should not need to use this code. Additionally, your usage does not look correct. Where are you getting this example from?

By default, the hardware back button listener in Ionic does not call multiple hardware back button events:

```js
document.addEventListener('ionBackButton', (ev) => {
  ev.detail.register(10, () => {
    console.log('Handler was called!');
  });
});
```

Instead, you can call the `processNextHandler` parameter to do this:

```js
document.addEventListener('ionBackButton', (ev) => {
  ev.detail.register(10, (processNextHandler) => {
    console.log('Handler was called!');
    processNextHandler();
  });
});
```

Does this address your concern?

**Comment by niallduffy12** [comment]
I am using a custom built modal as I could not use the ionic version for the design I required. I was trying to prevent the default action of the back button navigation. The closing of the modal works fine. But it closed both the modal and navigated back to the root. Sorry if my question was confusing. The above code actually solved the closing of the modal consistently without navigating back to the root but now the navigation won't work at all after a modal has been closed.
